backend/src/main/java/com/benine/backend/http/PresetCreationHandler.java

```diff
@@ -0,0 +1,91 @@
+package com.benine.backend.http;
+
+import com.benine.backend.LogEvent;
+import com.benine.backend.Main;
+import com.benine.backend.camera.Camera;
+import com.benine.backend.camera.CameraConnectionException;
+import com.benine.backend.camera.CameraController;
+import com.benine.backend.camera.ipcameracontrol.IPCamera;
+import com.benine.backend.database.DatabasePreset;
+import com.sun.net.httpserver.HttpExchange;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.jar.Attributes;
+
+/**
+ * Class allows creation of a preset by tagging a camera viewpoint location.
+**/
+public class PresetCreationHandler  extends RequestHandler {
+  
+ 
+  /**
+   * Create a new handler for creating new presets.
+   * @param controller the controller to interact with.
+   */
+  public PresetCreationHandler(CameraController controller) {
+    super(controller);
+  }
+  
+  /**
+     * Handles a request of making a new preset. 
+     * @param exchange the exchange containing data about the request.
+     * @throws IOException when an error occurs with responding to the request.
+  */
+  public void handle(HttpExchange exchange) throws IOException {
+    Attributes parsedURI;
+    try {
+      parsedURI = parseURI(exchange.getRequestURI().getQuery());
+        
+      int cameraID = Integer.parseInt(parsedURI.getValue("id"));
+      Camera camera =  getCameraController().getCameraById(cameraID);
+      IPCamera ipCamera = null;
+      
+      if (camera instanceof IPCamera) {
+        ipCamera = (IPCamera)camera;
```

Comment:
You can remove line 42 and change this line (line 45) to IPCamera ipCamera = ....